### PR TITLE
[Bugfix 1835135] sync to set default password save type of input dialog on running the application.

### PR DESCRIPTION
**Pull request by shenqianjin**
[AB#1835135](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1835135)

### CODE REVIEW

**Comment by andxu** [comment]
Please attach the final snapshot for UI change in this PR

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/connector/database/component/PasswordDialog.java [hunk 1]
```diff
@@ -58,10 +60,17 @@ public PasswordDialog(Project project, DatabaseResource resource) {
...
         testResultTextPane.setText(StringUtils.EMPTY);
-        final Dimension lastColumnSize = new Dimension(106, 30);
-        passwordSaveComboBox.setPreferredSize(lastColumnSize);
-        passwordSaveComboBox.setMaximumSize(lastColumnSize);
-        passwordSaveComboBox.setSize(lastColumnSize);
+        final Dimension lastColumnSize = new Dimension(106, this.passwordSaveComboBox.getPreferredSize().height);
+        this.passwordSaveComboBox.setPreferredSize(lastColumnSize);
+        this.passwordSaveComboBox.setMaximumSize(lastColumnSize);
+        this.passwordSaveComboBox.setSize(lastColumnSize);
+        if (Objects.nonNull(resource.getPassword()) && Objects.nonNull(resource.getPassword().saveType())) {
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/connector/database/component/PasswordDialog.java:67]: using Optional ofNullable to calculate saveType first and then there will be only one call to this.passwordSaveComboBox.setValue

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/connector/database/component/PasswordDialog.java [hunk 2]
```diff
@@ -58,10 +60,17 @@ public PasswordDialog(Project project, DatabaseResource resource) {
...
+            this.passwordSaveComboBox.setValue(resource.getPassword().saveType());
+        } else {
+            this.passwordSaveComboBox.setValue(Arrays.stream(Password.SaveType.values())
+                    .filter(e -> StringUtils.equals(e.name(), AzureConfigurations.getInstance().passwordSaveType())).findAny()
+                    .orElse(Password.SaveType.UNTIL_RESTART));
+        }
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/connector/database/component/PasswordDialog.java:73]: is this issue important/severe enough to be fixed as a hotfix?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/connector/database/component/PasswordDialog.java [hunk 3]
```diff
@@ -58,10 +60,17 @@ public PasswordDialog(Project project, DatabaseResource resource) {
...
-        passwordSaveComboBox.setMaximumSize(lastColumnSize);
-        passwordSaveComboBox.setSize(lastColumnSize);
+        final Dimension lastColumnSize = new Dimension(106, this.passwordSaveComboBox.getPreferredSize().height);
+        this.passwordSaveComboBox.setPreferredSize(lastColumnSize);
+        this.passwordSaveComboBox.setMaximumSize(lastColumnSize);
+        this.passwordSaveComboBox.setSize(lastColumnSize);
+        if (Objects.nonNull(resource.getPassword()) && Objects.nonNull(resource.getPassword().saveType())) {
+            this.passwordSaveComboBox.setValue(resource.getPassword().saveType());
+        } else {
+            this.passwordSaveComboBox.setValue(Arrays.stream(Password.SaveType.values())
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/connector/database/component/PasswordDialog.java:70]: this.passwordSaveComboBox.setValue(getSaveTypeOrDefault(resource))

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/connector/database/component/PasswordDialog.java (hunk 2 above)
Comment (wangmingliang-ms) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/connector/database/component/PasswordDialog.java:73]: should modify in `setData`
https://github.com/microsoft/azure-tools-for-java/pull/5622/files#diff-6005d74a14cff22bf7a458fab4d67ded17c3e7112649afe323ccf6691e01b1dcR161-R163

Comment (shenqianjin) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/connector/database/component/PasswordDialog.java]: updated.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/connector/database/component/PasswordDialog.java (hunk 3 above)
Comment (shenqianjin) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/connector/database/component/PasswordDialog.java]: updated.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/connector/database/component/PasswordDialog.java [hunk 4]
```diff
@@ -159,7 +153,9 @@ public Password getData() {
 
     @Override
     public void setData(Password data) {
-        passwordSaveComboBox.setValue(data.saveType());
+        this.passwordSaveComboBox.setValue(Optional.ofNullable(data).map(e -> e.saveType()).orElse(Arrays.stream(Password.SaveType.values())
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/connector/database/component/PasswordDialog.java:156]: lambda

Comment (shenqianjin) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/connector/database/component/PasswordDialog.java:156]: updated.

**Review by wangmingliang-ms — APPROVED** [review]

**Pull request merged**